File: php/StringCalculator.php
<?php
class StringCalculator
{
  public function add($numberString) {
    if ($numberString === 0) {
      return 0;
    }

    $containsCustomDelimiter = strpos($numberString, "//");

    if ($containsCustomDelimiter === 0) {
      $delimiter = substr($numberString, 2, 1);
      $numberString = substr($numberString, 4);
      $numberArray = explode($delimiter, $numberString);
    } else {
      $numberArray = preg_split("/[\n,]+/", $numberString);
    }

    $sum = 0;
    foreach ($numberArray as $number) {
      $sum += (int) $number;
    }
    return $sum;
  }
}
?>

File: python/string_calculator.py
import re


class StringCalculator:
    def add(self, number_string):
        if number_string == "":
            return 0

        if number_string.startswith("//"):
            delimiter = number_string[2]
            number_string = number_string[4:]
            number_array = number_string.split(delimiter)
        else:
            number_array = re.split(r"[\n,]+", number_string)

        return sum(int(n) for n in number_array)
